source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm

```diff
@@ -365,28 +365,54 @@ + (UIColor *)convertHexColorCodeToUIColor:(const std::string&)hexColorCode
     return color;
 }
 
-- (UIColor *)getBackgroundColorForContainerStyle:(ACRContainerStyle)style
+- (ContainerStyleDefinition) getColorPalette:(ACRContainerStyle)style
 {
-    const std::string &hexColorCode = (style == ACREmphasis)?
-        _config->GetContainerStyles().emphasisPalette.backgroundColor :
-        _config->GetContainerStyles().defaultPalette.backgroundColor;
+    switch (style) {
+        case ACRAccent:
+            return _config->GetContainerStyles().accentPalette;
+        case ACRAttention:
+            return _config->GetContainerStyles().attentionPalette;
+        case ACREmphasis:
+            return _config->GetContainerStyles().emphasisPalette;
+        case ACRGood:
+            return _config->GetContainerStyles().goodPalette;
+        case ACRWarning:
+            return _config->GetContainerStyles().warningPalette;
+        case ACRDefault:
+        default:
+            return _config->GetContainerStyles().defaultPalette;
+    }
+}
 
+- (UIColor *)getBackgroundColorForContainerStyle:(ACRContainerStyle)style
+{
+    std::string hexColorCode = [self getColorPalette:style].backgroundColor;
     return [ACOHostConfig convertHexColorCodeToUIColor:hexColorCode];
 }
 
 + (ACRContainerStyle)getPlatformContainerStyle:(ContainerStyle)style
 {
-    ACRContainerStyle containerStyle = ACRDefault;
+    ACRContainerStyle containerStyle;
```

Comment:
why did you remove the initialization with the default value?

Reply:
The default value is assigned in the `default` case for the switch-case. Felt it would be redundant to initialize it, but should I change it?

Reply:
yes, assignment takes only 2-3 instructions.

---
In reply to: [249849267](https://github.com/Microsoft/AdaptiveCards/pull/2289#discussion_r249849267) [](ancestors = 249849267)